lib/ansible/modules/clustering/etcd3.py

```diff
@@ -107,9 +160,13 @@ def run_module():
     # supports check mode
     module = AnsibleModule(
         argument_spec=module_args,
-        supports_check_mode=True
+        supports_check_mode=True,
+        required_together=[['client_cert', 'client_key'], ['user', 'password']],
     )
 
+    if module.params['ca_cert'] is None and module.params['client_cert'] is not None:
```

Comment:
doesn't `required_together` take care of this?

Reply:
Ah, okay. Makes sense then.

Reply:
why don't you check `client_key` also?

Reply:
I could, but it would not change anything, since `required_together` already ensures that `client_cert` and `client_key` are always present or absent together.
Do you think it would be more readable with both requirements written?

Reply:
I guess. It wasn't obvious to me. At least a comment needed.

Reply:
Indeed, it is done;)